network/components/virt_rx.c

```diff
@@ -92,24 +106,44 @@ void rx_return(void)
             // [2]: https://developer.arm.com/documentation/100236/0002/functional-description/cache-behavior-and-cache-protection/invalidating-or-cleaning-a-cache
             cache_clean_and_invalidate(buffer_vaddr, buffer_vaddr + ROUND_UP(buffer.len, 1 << CONFIG_L1_CACHE_LINE_SIZE_BITS));
 
-            int client = get_client((struct ethernet_header *) buffer_vaddr);
-            if (client >= 0) {
-                err = net_enqueue_active(&state.rx_queue_clients[client], buffer);
-                assert(!err);
-                notify_clients[client] = true;
+            bool is_broadcast_packet = is_broadcast((struct ethernet_header *) buffer_vaddr);
```

Comment:
We should be able to perform the `get_client` operation and `is_broadcast` operation in the same function call... No need to check the MAC address twice...